lib/ACL/Rule.php

```diff
@@ -145,7 +145,7 @@ public static function mergeRules(array $rules): Rule {
 		}, 0);
 		// or'ing the permissions combines them with allow overwriting deny
 		$permissions = array_reduce($rules, function (int $permissions, Rule $rule) {
-			return $permissions | $rule->getPermissions();
+			return $permissions | ($rule->getPermissions() & $rule->getMask());
```

Comment:
Reading comments on the constructor, should the mask be applied to permissions in constructor to be safe? It states that `$this->permissions` is supposed to only apply to non-inherited values, so it should only allow it to contain permission for those, no?